ads/inabox/inabox-messaging-host.js

```diff
@@ -122,15 +122,24 @@ export class InaboxMessagingHost {
       return false;
     }
 
-    const iframe =
+    const adFrame =
         this.getFrameElement_(message.source, request['sentinel']);
-    if (!iframe) {
+    if (!adFrame) {
       dev().info(TAG, 'Ignored message from untrusted iframe:', message);
       return false;
     }
 
+    const allowedTypes = adFrame.iframe.dataset['ampAllowed'];
+    const hasAllowedTypeRegex =
+        new RegExp(`(^|,)\\s*${request['type']}\\s*($|,)`);
```

Comment:
`request['type']` is untrusted, so we should escape it before including it in the regexp: 

```
Escaping user input that is to be treated as a literal string within
a regular expression—that would otherwise be mistaken for a
special character—can be accomplished by simple replacement:

function escapeRegExp(string) {
  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
}
```
-- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

Reply:
Just did. PTAL and see if it's correct.

Reply:
looks good!